includes/GbWeiss.php

```diff
@@ -138,9 +140,48 @@ public function validateCredentials(): void
         }
     }
 
+    /**
+     *  Requests a new OAuthToken and stores the accessToken in
+     *  the ws_options table
+     *
+     *  @throws \Exception If the token could not be saved.
+     */
+    public function updateAuthToken(): bool
+    {
+        $clientId = get_option('gbw_client_id', false);
+        $clientSecret = get_option('gbw_client_secret', false);
+
+        try {
```

Comment:
If you do nothing else than throwing an exception in the catch block you could also not catch the exception. It does not make a difference.